lib/bitex_bot/models/api_wrappers/kraken/kraken_api_wrapper.rb

```diff
@@ -1,62 +1,67 @@
 # Wrapper implementation for Kraken API.
 # https://www.kraken.com/en-us/help/api
 class KrakenApiWrapper < ApiWrapper
+  attr_accessor :api_key, :api_secret, :client
+
   MIN_AMOUNT = 0.002
 
-  def self.setup(settings)
-    HTTParty::Basement.headers('User-Agent' => BitexBot.user_agent)
-    @settings = settings
+  def initialize(settings)
+    self.api_key = settings.api_key
+    self.api_secret = settings.api_secret
+    setup
   end
 
-  def self.client
-    @client ||= KrakenClient.load(@settings)
+  def setup
+    KrakenOrder.api_wrapper = self
```

Comment:
KrakenOrder need know who is your wrapper, this makes it not multi-client. It needs to be rethought.